src/components/Map/__tests__/Map.spec.tsx

```diff
@@ -23,84 +25,123 @@ describe('<Map/> that renders a Map with markers', () => {
     });
+
     it('should fit map to markers', () => {
-        mount(<Map markers={[pointMarker, regionMarker]} />);
+        render(<Map markers={[pointMarker, regionMarker]} />);
         // call fitBounds for each marker
         expect(fitBoundsMock).toHaveBeenCalledTimes(1);
     });
+
     it('should fit map when new marker is added', () => {
-        const wrapper = mount(<Map markers={[pointMarker]} />);
+        const view = render(<Map markers={[pointMarker]} />);
+
         expect(fitBoundsMock).toHaveBeenCalledTimes(1);
-        wrapper.setProps({ markers: [pointMarker, regionMarker] });
+
+        view.rerender(
+            <Map
+                defaultHighlight={NL_PATHS as GeoJSON.GeoJsonObject}
+                markers={[pointMarker, regionMarker]}
+            />
+        );
+
         // call count = original call + again for each marker + 1 more because markers are updated with a small delay
         expect(fitBoundsMock).toHaveBeenCalledTimes(3);
     });
+
     it('should fit to default center and zoom if markers removed', () => {
-        const wrapper = mount(<Map markers={[pointMarker, regionMarker]} />);
-        wrapper.setProps({ markers: [] });
+        const view = render(<Map markers={[pointMarker, regionMarker]} />);
+        view.rerender(<Map defaultHighlight={NL_PATHS as GeoJSON.GeoJsonObject} markers={[]} />);
+
         expect(setZoomMock).toHaveBeenCalledTimes(1);
         expect(setCenterMock).toHaveBeenCalledTimes(1);
     });
+
     it('should add highlighted areas when geoJson objects are passed as markers', () => {
-        mount(<Map markers={[FR_FRIESLAND]} />);
+        render(<Map markers={[FR_FRIESLAND as RegionArea]} />);
+
         expect(addGeoJsonMock).toHaveBeenCalledTimes(1);
         expect(setZoomMock).toHaveBeenCalledTimes(1);
         expect(setCenterMock).toHaveBeenCalledTimes(1);
     });
+
     it('should recenter the map and change zoom when geoJson objects are replaced by regular markers', () => {
-        const wrapper = mount(<Map markers={[FR_FRIESLAND]} />);
+        const view = render(<Map markers={[FR_FRIESLAND as RegionArea]} />);
+
         expect(setZoomMock).toHaveBeenCalledTimes(1);
         expect(setCenterMock).toHaveBeenCalledTimes(1);
-        wrapper.setProps({ markers: [pointMarker, regionMarker] });
+
+        view.rerender(
+            <Map
+                defaultHighlight={NL_PATHS as GeoJSON.GeoJsonObject}
+                markers={[pointMarker, regionMarker]}
+            />
+        );
+
         expect(setZoomMock).toHaveBeenCalledTimes(2);
         expect(setCenterMock).toHaveBeenCalledTimes(2);
     });
+
     it('should add default highlight when geoJson markers are reset', () => {
-        const wrapper = mount(<Map defaultHighlight={NL_PATHS} markers={[FR_FRIESLAND]} />);
+        const view = render(
+            <Map
+                defaultHighlight={NL_PATHS as GeoJSON.GeoJsonObject}
+                markers={[FR_FRIESLAND as FeatureCollection]}
```

Comment:
I'm confused about these types.. I'd think both these files are GeoJSON files, so why we cast them differently. Also a few lines above it reads `FR_FRIESLAND as RegionArea`. Can these be more aligned, or clear when we use which?

Reply:
I'm also confused about all these types. What I got from the [documentation](https://www.grapecity.com/wijmo/api/interfaces/wijmo_chart_map.geojson.geojsonobject.html):

- `GeoJsonObject` is the base type for GeoJson that includes other types `FeatureCollection `, `Point`, `MultiPoint` etc.
-  `FeatureCollection` is the list of [features](https://rdrr.io/cran/geoops/man/Feature.html). 

Probably for `markers` prop, we need only `FeatureCollection` type that can draw the circles by coordinates. 
We can create a follow-up task to investigate it later.
Also, I removed `geojson` library. We can use `GeoJSON.FeatureCollection` instead.